### PR TITLE
Add `typecheck.macro`

### PR DESCRIPTION
**Pull request by vedantroy**
Add typecheck.macro (https://github.com/vedantroy/typecheck.macro).

Note: I added a "Typescript" section because typecheck.macro is a macro that specifically only works with Typescript.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -62,6 +63,9 @@
 - [`traph.macro`](https://github.com/caesarsol/traph.macro): Transform Objects easily, leveraging object getters and graphs
 - [`idx.macro`](https://github.com/dralletje/idx.macro): Traverse properties on objects and arrays
 
+### Typescript
```
Comment (jgierer12) [inline readme.md]: Please use the correct capitalization (`TypeScript`) here as well as in the table of contents.

If we're adding a TS section we might as well move `ts-nameof` in there from the General section.

File: readme.md [hunk 2]
```diff
@@ -62,6 +63,9 @@
...
+- [`typecheck.macro`](https://github.com/vedantroy/typecheck.macro): Automatically generate type-checker functions for Typescript types.
```
Comment (jgierer12) [inline readme.md]:
```suggestion
- [`typecheck.macro`](https://github.com/vedantroy/typecheck.macro): Automatically generate type-checker functions for TypeScript types
```

Just for consistency

**Review by jgierer12 — CHANGES_REQUESTED** [review]
Thanks, just need some minor changes before I can merge this 🙂

**Comment by jgierer12** [comment]
By the way, I haven't read your whole readme yet but I didn't see a before/after example of the macro output ([like this](https://github.com/kentcdodds/import-all.macro#usage)). That's always one of the first things I look for when looking at a new macro to help me understand what it really does. You might want to consider putting something like that near the top of your readme.

**Comment by vedantroy** [comment]
@jgierer12 I addressed your feedback!

**Comment by jgierer12** [comment]
Thanks 🙌

**Pull request merged**